### PR TITLE
🚀 Remove IE specific references for ESM path

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Remove IE specific workaround called out in the module output.

### CODE REVIEW

File: src/service/navigation.js [hunk 1]
```diff
@@ -624,16 +624,18 @@ export class Navigation {
...
     // @see https://github.com/ampproject/amphtml/issues/18671
-    if (Services.platformFor(this.ampdoc.win).isIe()) {
-      const id = toLocation.hash.substring(1);
-      const elementWithId = this.ampdoc.getElementById(id);
-      if (elementWithId) {
-        if (
-          !/^(?:a|select|input|button|textarea)$/i.test(elementWithId.tagName)
-        ) {
-          elementWithId.tabIndex = -1;
+    if (!IS_ESM) {
```
Comment (jridgewell) [inline src/service/navigation.js:627]: Can we inline this into `isIe`? That way every call site can just be fixed?

Comment (kristoferbaxter) [reply src/service/navigation.js:627]: Originally that's what I did, but there are a lot more callers of `isIe` and was too timid to make such a large change without the tests running fully on `esm` yet.

Comment (kristoferbaxter) [reply src/service/navigation.js]: Resolved with #d61b798.

**Comment by erwinmombay** [comment]
force merged because of misbehaving bot per @kristoferbaxter

**Pull request merged**